### PR TITLE
Add `search_endpoint` type and `unique` value parameters to search input fields (relation widget)

### PR DESCRIPTION
**Pull request by volterra79**
Fix issue on select input type bind to value relation type

### CODE REVIEW

**Comment by Raruto** [comment]
@volterra79 could you please add some more details to this pull? (what problem does it solve? what solution has been adopted? what is a relation? ...)

As usual the screenshots help a lot, but remember to also report the text of any errors shown inside (frontend interface or browser console), otherwise it is practically impossible to search for them in github...

**Comment by volterra79** [comment]
@Raruto ~~the issue was related to search select input remain disabled as follow: [Screenshot from 2022-10-26 12-01-52](https://user-images.githubusercontent.com/1051694/197997914-3dc399d6-781d-4b62-b025-6301af041e5f.png)~~

(see: https://github.com/g3w-suite/g3w-client/pull/227#issuecomment-1292177482)

**Comment by Raruto** [comment]
> @Raruto the issue was related to search select input remain disabled 

@volterra79 ok, but how does it differ from this other one? https://github.com/g3w-suite/g3w-client/pull/240

**Comment by volterra79** [comment]
@Raruto in #240 this change  https://github.com/g3w-suite/g3w-client/pull/240/files#diff-c77c5501ee3b1edd69511fdfa9147a904685920b2a25b109c116850b9b420665 is related to this issue solved by pull request.
Plus change the way that v-disable directive for application purpose

**Comment by Raruto** [comment]
@volterra79 please explain better what is the purpose of: https://github.com/g3w-suite/g3w-client/pull/227, because these two bug "reports" look identical to me..

- https://github.com/g3w-suite/g3w-client/pull/227#issuecomment-1291797075 --> solved by [#227](https://github.com/g3w-suite/g3w-client/pull/227)
- https://github.com/g3w-suite/g3w-client/issues/239 --> solved by [#240](https://github.com/g3w-suite/g3w-client/pull/240)

**Comment by volterra79** [comment]
@Raruto my mistake. #227 is a refactor of `createSingleFieldParameter` method of `utils.js` to get as parameter also `search_endpoint`.

**Comment by Raruto** [comment]
@volterra79 so is this a bug? a code refactoring? a feature request? a ux change? (just by reading the code, it's hard to understand what's the purpose...)

File: src/app/core/utils/utils.js [hunk 1]
```diff
@@ -474,16 +474,27 @@ const utils = {
...
-    logicop = logicop && `|${logicop}`;
-    if (Array.isArray(value)){
-      let filter = '';
-      const valueLenght = value.length;
-      value.forEach((value, index) =>{
-        filter+=`${field}|${operator}|${encodeURIComponent(value)}${index < valueLenght - 1 ? `${logicop},` : ''}`
-      });
-      return filter
-    } else return `${field}|${operator.toLowerCase()}|${encodeURIComponent(value)}${logicop || ''}`;
+  createSingleFieldParameter({field, value, operator='eq', logicop=null, search_endpoint="api"}){
```
Comment (volterra79) [inline src/app/core/utils/utils.js:477]: Add new argument `search_endpoint` used to create different filter based on type of search_endpoint value:
```
api: g3w-admin api
ows: wms filter parameter
```

File: src/app/gui/search/vue/panel/searchservice.js [hunk 2]
```diff
@@ -205,8 +205,25 @@ proto.createFieldsDependenciesAutocompleteParameter = function({fields=[], field
...
-proto.getValuesFromField = function(field){
-  if (field.options.layer_id) return this.getValueRelationValues(field);
+proto.getValuesFromField = async function(field){
+  if (field.options.layer_id) {
+    const uniqueValues = await this.getUniqueValuesFromField({
+      field,
+      unique: field.options.key
+    });
+    const layer = CatalogLayersStorRegistry.getLayerById(field.options.layer_id);
+    const filter = createFilterFormInputs({
```
Comment (volterra79) [inline src/app/gui/search/vue/panel/searchservice.js:215]: add a `search_endpoint` value to create right filter

File: src/app/gui/search/vue/panel/searchservice.js [hunk 3]
```diff
@@ -205,8 +205,25 @@ proto.createFieldsDependenciesAutocompleteParameter = function({fields=[], field
  * @param field (form input)
  * @returns {Promise<*[]>}
  */
-proto.getValuesFromField = function(field){
-  if (field.options.layer_id) return this.getValueRelationValues(field);
+proto.getValuesFromField = async function(field){
+  if (field.options.layer_id) {
+    const uniqueValues = await this.getUniqueValuesFromField({
+      field,
+      unique: field.options.key
```
Comment (volterra79) [inline src/app/gui/search/vue/panel/searchservice.js:212]: add `unique` value to add query parameter to server request

**Pull request merged**